channeldb/kvdb/etcd/stm.go

```diff
@@ -711,7 +701,12 @@ func (s *stm) OnCommit(cb func()) {
 // because the keys have changed return a CommitError, otherwise return a
 // DatabaseError.
 func (s *stm) commit() (CommitStats, error) {
-	rset := s.rset.cmps(s.lset)
+	// Remove excluded keys from the read set.
+	for key := range s.xset {
+		delete(s.rset, key)
```

Comment:
it's clear what this does, but it's not totally obvious _why_ we would do this. if the bucket is accessed during the transaction, it's not clear to me why it should be removed from the read set and how we can guarantee consistency with this change.

is this commit necessary to correct the range iteration? seems maybe like an orthogonal optimization?

Reply:
Discussed this offline. To summarize: exclude set while in theory would reduce the cmp set and thereby make transactions less likely to collide are dangerous when another transaction tries to delete a bucket that is excluded in the current transaction. As a conclusion I removed the exclude set concept completely (while still keeping the changes for removing the lock set which is still wrong due to increasing contention on top level buckets).